apps/core/scenes/results/ResultsListItem.js

```diff
@@ -11,51 +13,234 @@ type Props = {|
+export type CarrierData = {
+  code: string,
+  name: string,
+  type?: 'airline' | 'bus' | 'train',
 };
 
-const styles = StyleSheet.create({
-  listItem: {
-    fontSize: 12,
-    padding: 2,
-    margin: 2,
-    backgroundColor: defaultTokens.backgroundBody,
-  },
-});
+export type TripSectorProps = {
+  +id: string,
+  +arrival: string,
+  +arrivalTime: string,
+  +carrier: CarrierData,
+  +tripDate: string,
+  +departure: string,
+  +departureTime: string,
+  +duration: string,
+};
+
+export type Route = $PropertyType<ResultsListItemType, 'route'>;
+export type RouteItem = $ElementType<$NonMaybeType<Route>, number>; // number because arrays are number-indexed
+
+const dateFormat = 'ccc d LLL';
+
+class ResultListItem extends React.Component<Props> {
+  getDateTimeInterval = (
+    to: DateTime,
+    from: DateTime,
+    unit: string | string[] = ['hours', 'minutes'],
+  ) => {
+    return to.diff(from, unit).toObject();
+  };
+
+  intervalToString = (interval: Duration, type: 'days' | 'hours' = 'hours') => {
+    if (type === 'days') {
+      switch (interval.days) {
+        case 0:
+          return `${Math.floor(interval.hours)} hours`;
+        case 1:
+          return `${interval.days} day`;
+        default:
+          return `${interval.days} days`;
+      }
+    }
+    return (
+      (interval.hours ? interval.hours + 'h' : '') +
+      (interval.minutes ? ' ' + interval.minutes + 'm' : '')
+    );
+  };
+
+  // @TODO Handle cases when we don't have some data from GraphQL/REST server
+  getSanitizedRouteItem = (route: RouteItem): ?TripSectorProps => {
+    if (route) {
+      const localArrival = DateTime.fromISO(route.localArrival);
+      const localDeparture = DateTime.fromISO(route.localDeparture);
+      const utcArrival = DateTime.fromISO(route.utcArrival);
+      const utcDeparture = DateTime.fromISO(route.utcDeparture);
+      return {
+        arrival: route.cityTo ?? '',
+        arrivalTime:
+          (localArrival && localArrival.toLocaleString(DateTime.TIME_SIMPLE)) ??
+          '',
+        carrier: { code: route.airline ?? '', name: '' }, // @TODO: get name of the airline
+        tripDate: (localDeparture && localDeparture.toFormat(dateFormat)) ?? '',
+        departure: route.cityFrom ?? '',
+        departureTime:
+          (localDeparture &&
+            localDeparture.toLocaleString(DateTime.TIME_SIMPLE)) ??
+          '',
+        duration: this.intervalToString(
+          this.getDateTimeInterval(utcArrival, utcDeparture),
+        ),
+        id: route.id ?? '',
+      };
+    }
+    return null;
+  };
+
+  getSectorBorderIndex = (): ?number => {
+    const { route, routes } = this.props.data;
+    if (!route) {
+      return null;
+    }
+    // @TODO polish awful condition if it's possible
+    return route.findIndex(
+      routeItem =>
+        routeItem &&
+        routes &&
+        routes.length === 2 &&
+        routes[1] &&
+        routeItem.flyFrom === routes[1][0],
```

Comment:
You can probably use optional chaining here
```
routeItem => routeItem?.flyFrom === routes?.[1]?.[0]
```

Should be the same (I think)

Reply:
Wow 👍 what is this syntax? I haven't seen it before.

Reply:
It's called optional chaining
https://github.com/tc39/proposal-optional-chaining